go/vt/vtgate/planbuilder/abstract/join.go

```diff
@@ -58,8 +58,7 @@ func (j *Join) PushPredicate(expr sqlparser.Expr, semTable *semantics.SemTable)
 			j.LeftJoin = false
 			return j.RHS.PushPredicate(expr, semTable)
 		}
-		semTable.ProjectionErr = vterrors.Errorf(vtrpcpb.Code_UNIMPLEMENTED, "unsupported: cross-shard left join and where clause")
-		return nil
```

Comment:
Why revert this change? Isn't it okay to keep it as a `ShardedErr`?

Reply:
Well, we have to either return an error, or build something that adds the predicate to the operator tree. At the moment we don't have a filter operator, so this will be easy to fix post-remove-query-trees